snappy-sys/build.rs

```diff
@@ -45,61 +38,37 @@ fn configure_snappy(want_static: bool) -> bool {
 }
 
 fn build_snappy() {
-    let src = PathBuf::from(&env::var("CARGO_MANIFEST_DIR").unwrap()).join("snappy");
-    let build = PathBuf::from(&env::var("OUT_DIR").unwrap()).join("build");
+    let out_dir = PathBuf::from(&env::var("OUT_DIR").unwrap());
 
-    let cc = gcc::Config::new().get_compiler();
+    let cc = cc::Build::new().get_compiler();
     let mut cflags = OsString::new();
     for arg in cc.args() {
         cflags.push(arg);
         cflags.push(" ");
     }
-    t!(fs::create_dir_all(&build));
-    cp_r(&src, &build);
-    run(Command::new("./autogen.sh").current_dir(&build));
-    run(Command::new("./configure")
+    cmake::Config::new("snappy")
         .env("CC", cc.path())
         .env("CFLAGS", cflags)
-        .current_dir(&build)
-        .arg("--disable-shared")
-        .arg("--enable-static")
-        .arg("--with-pic"));
-    run(Command::new("make").current_dir(&build));
+        .env("CMAKE_BUILD_TYPE", "static")
+        .build();
     println!("cargo:rustc-link-lib=static=snappy");
-    println!("cargo:rustc-link-search={}/.libs", build.to_string_lossy());
-    println!("cargo:root={}", build.to_string_lossy());
+    println!("cargo:rustc-link-search={}", out_dir.join("lib").to_string_lossy());
+    println!("cargo:root={}", out_dir.to_string_lossy());
```

Comment:
You might want to add "=native=" to "link-search" just to be explicit that this path is only for native libs.